samples/dfu/src/flash.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2021 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(golioth_dfu);
+
+#include <stdio.h>
+#include <storage/flash_map.h>
+
+#include "flash.h"
+
+/* FLASH_AREA_ID() values used below are auto-generated by DT */
+#ifdef CONFIG_TRUSTED_EXECUTION_NONSECURE
+#define FLASH_AREA_IMAGE_PRIMARY FLASH_AREA_ID(image_0_nonsecure)
+#define FLASH_AREA_IMAGE_SECONDARY FLASH_AREA_ID(image_1_nonsecure)
+#else
+#define FLASH_AREA_IMAGE_PRIMARY FLASH_AREA_ID(image_0)
+#if FLASH_AREA_LABEL_EXISTS(image_1)
+#define FLASH_AREA_IMAGE_SECONDARY FLASH_AREA_ID(image_1)
+#endif
+#endif /* CONFIG_TRUSTED_EXECUTION_NONSECURE */
+
+char current_version_str[sizeof("255.255.65535")];
+
+static int current_version_init(const struct device *dev)
+{
+	struct mcuboot_img_header hdr;
+	int written;
+	int err;
+
+	err = boot_read_bank_header(FLASH_AREA_IMAGE_PRIMARY, &hdr, sizeof(hdr));
+	if (err) {
+		LOG_ERR("Failed to read primary area (%u) header: %d",
+			FLASH_AREA_IMAGE_PRIMARY, err);
+		return err;
+	}
+
+	written = snprintf(current_version_str, sizeof(current_version_str),
+			   "%u.%u.%u",
+			   (unsigned int) hdr.h.v1.sem_ver.major,
+			   (unsigned int) hdr.h.v1.sem_ver.minor,
+			   (unsigned int) hdr.h.v1.sem_ver.revision);
+	if (written >= sizeof(current_version_str)) {
+		LOG_ERR("Version string is too long!");
+		current_version_str[0] = '\0';
+		return -ENOMEM;
+	}
+
+	return 0;
+}
+
+SYS_INIT(current_version_init, APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY);
+
+/**
+ * Determines if the specified area of flash is completely unwritten.
+ *
+ * @note This is a copy of zephyr_img_mgmt_flash_check_empty() from mcumgr.
+ */
+static int flash_area_check_empty(const struct flash_area *fa,
+				  bool *out_empty)
+{
+	uint32_t data[16];
+	off_t addr;
```

Comment:
feel free to be verbose with variable names. eg "flashAreaAddress"

Reply:
This variable is used in function called `flash_area_check_empty()`, so it already gives a context of what `addr` is about. I don't see a value in using `flash_area_` prefix.

Anyway, as noted in comment above function body, this is copy pasted from mcuboot code, with the same variable names. Or rather an older version of it, as I can see they have updated implementation in the meantime.

Reply:
My brain doesn't like to assume. I also like that it provides some context when looking at the signature of the function and better self documentation.

It's approachable to new users of a codebase.

I'm not going to enforce but keep the newbs in mind.